lib/ukrand/random.c

```diff
@@ -0,0 +1,89 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/random.h>
+#include <string.h>
+
+static int uk_hwrand_init(struct uk_init_ctx *ictx __unused)
+{
+	return ukarch_random_init();
+}
+uk_early_initcall(uk_hwrand_init, 0x0);
+
+#if defined CONFIG_LIBUKRAND_HW
+int uk_random_u32(__u32 *val)
+{
+	return ukarch_random_u32(val);
+}
+
+int uk_random_u64(__u64 *val)
+{
+	return ukarch_random_u64(val);
+}
+
+int uk_random_seed_u32(__u32 *val)
+{
+	return ukarch_random_seed_u32(val);
+}
+
+int uk_random_seed_u64(__u64 *val)
+{
+	return ukarch_random_seed_u64(val);
+}
+
+#elif defined CONFIG_LIBUKRAND_CHACHA
+int uk_random_u32(__u32 *val)
```

Comment:
How about we place the `#if`'s inside the functions themselves, to reduce code duplication. E.g.
```C
int uk_random_seed_u64(__u64 *val)
{
       int rc = 0;
       
#if CONFIG_LIBUKRAND_HW
	rc = ukarch_random_seed_u64(val);
#elif CONFIG_LIBUKRAND_CHACHA
       *val = uk_chacha_randr_u64();
 #else /* !CONFIG_LIBUKRAND_HW && !CONFIG_LIBUKRAND_CHACHA */
 #error "Invalid RNG source"
 #endif /* !CONFIG_LIBUKRAND_HW && !CONFIG_LIBUKRAND_CHACHA */
 
       return rc;
}
```

Reply:
Its cleaner this way. Thanks!